### PR TITLE
add a blog post about blackfire-extension

### PR DESCRIPTION
**Pull request by agallou**

### CODE REVIEW

File: _posts/2016-02-08-blackfire-extension.markdown [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+author: "@atoum/all"
+categories: description
+---
+
+Intro
+-----
+
+Since the release of its [version 2](http://blog.blackfire.io/blackfire-v2-0-automate-performance-testing.html), [Blackfire](blackfire.io) allows the automation of performance tests.
+
+blackfire provides a [PHP SDK](https://blackfire.io/docs/reference-guide/php-sdk) that, in other things, allows to integrate performance tests inside PHPUnit : you can check that your code doesn't exhausts a memory limit, an execution time or a number of calls for a function/method.
```
Comment (jubianchi) [inline _posts/2016-02-08-blackfire-extension.markdown]: `s/in other things/among other things/`

File: _posts/2016-02-08-blackfire-extension.markdown [hunk 2]
```diff
@@ -0,0 +1,138 @@
+---
+layout: post
+title: "Execute blackfire assertions inside atoum"
+date: 2016-02-08 09:00:00
+author: "@atoum/all"
```
Comment (jubianchi) [inline _posts/2016-02-08-blackfire-extension.markdown]: Put your own username here ;)

File: _posts/2016-02-08-blackfire-extension.markdown [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    ->addToRunner($runner)
+;
+{% endhighlight %}
+
+Usage
+-----
+
+The extension will provide a new asserter : `blackfire`.
+
+This methods can we called from this new asserter:
```
Comment (jubianchi) [inline _posts/2016-02-08-blackfire-extension.markdown]: `s/this methods/those methods/`

Comment (jubianchi) [inline _posts/2016-02-08-blackfire-extension.markdown]: `s/can we/can be/`

File: _posts/2016-02-08-blackfire-extension.markdown [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+The extension will provide a new asserter : `blackfire`.
+
+This methods can we called from this new asserter: 
+
+* assert
+* defineMetric
+* profile
+
+Let's take this test as example:
```
Comment (jubianchi) [inline _posts/2016-02-08-blackfire-extension.markdown]: `s/as example/as an example/`

File: _posts/2016-02-08-blackfire-extension.markdown [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+                    sleep(4);
+                    // call code to instrument
+                })
+        ;
+    }
+}
+
+{% endhighlight %}
+
+This test will check that the code between the anonymous function will not take more than 2 seconds to be executed.
```
Comment (jubianchi) [inline _posts/2016-02-08-blackfire-extension.markdown]: `s/between/inside/`

File: _posts/2016-02-08-blackfire-extension.markdown [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+
+Here we've added a `sleep(4)` to make the test fail, so an output like this will be displayed:
+
+![Output blackfire atoum](/images/posts/2016-02-08-blackfire-extension/screenshot.png)
+
+The callback has been called, instrumented by blackfire, sent to blackfire.io. blackfire.io has sent us the assertions results which have been integrated inside the unit tests results.
+
+This a an example and based on the execution time. You can define more usefull assertions like this:
```
Comment (jubianchi) [inline _posts/2016-02-08-blackfire-extension.markdown]: `s/This a an example and based/This is an example based/`

File: _posts/2016-02-08-blackfire-extension.markdown [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+./vendor/bin/atoum -d tests --filter 'not(extensions contains "blackfire")'
+```
+
+
+Last thoughts
+-------------
+
+This extension requires a leat the version 2.5.0 of atoum : ([read the 2.5.0 announcement](http://atoum.org/release/2016/01/08/release-2-5-0.html)).
+
+A lot of [metric](https://blackfire.io/docs/reference-guide/metrics) and [variables](https://blackfire.io/docs/reference-guide/assertions) are available out of the box. As we seen previousely, you can define your own metric. So don't hesitate to add those assertions in your unit tests.
```
Comment (Grummfy) [inline _posts/2016-02-08-blackfire-extension.markdown]: saw previously
or
have seen previously

**Comment by agallou** [comment]
errors/typos fixed. thanks @jubianchi @Grummfy

**Comment by jubianchi** [comment]
godd for me! should I deploy and wéll broadcast on Twitter this Minday ?

**Comment by agallou** [comment]
I let you choose.
The important thing is to not communicate on the week end.
The only thing is that the post will appear written in the future for a couple of days for the website visitors and RSS readers, but It don't seem to be a problem.

**Comment by jubianchi** [comment]
too bad we can't schedule a merge on github :)

I'm going to merge this so that on Monday, even if one of us is not ready to tweet it, anyone could tweet it ;)

**Pull request merged**